ci/deploy-vmck

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash -ex
+
+cd "$( dirname "${BASH_SOURCE[0]}" )"
+
+export VAGRANT_DOTFILE_PATH=$(mktemp -d --tmpdir VAGRANT_DOTFILE_XXXXXXXX)
+export VAGRANT_DEFAULT_PROVIDER=vmck
+export VAGRANT_CHECKPOINT_DISABLE=true
+export VAGRANT_BOX_UPDATE_CHECK_DISABLE=true
+
+TIMEOUT_MIN=23
+RETRIES=3
+
+function print_section() {
+  set +x
+  echo
+  echo '-----------------------------------------'
+  echo "| $1"
+  echo '-----------------------------------------'
+  set -x
+}
+
+function vagrant_up() {
+  set +e
+  vagrant up --no-provision || echo "vagrant up failed, VM might still work"
+  echo "sudo shutdown +$TIMEOUT_MIN" | vagrant ssh
```

Comment:
This `--no-provision` was a neat trick, to be able to call `shutdown +30` before running `provision`, but here's a cleaner approach: https://github.com/eaudeweb/ecolex-cluster/blob/98e136208a9f7b1555910d1fc4ddee1cd73b8ed8/Vagrantfile#L14 (in short, run `shutdown` as the first step in provisioning, if a particular env var is set). The idea is to run `vagrant up` in one step, instead of `vagrant up --no-provision` followed by `vagrant ssh` followed by `vagrant provision`.